### PR TITLE
fix search bar and makespin bug

### PR DESCRIPTION
**Pull request by Wilko2001**

### CODE REVIEW

**Comment by cma06** [comment]
We should look into these https://playwright.dev/docs/actionability rather than setting more timeouts in the code

File: skyVegasTests/utilities/makeSpinHelper.ts [hunk 1]
```diff
@@ -3,7 +3,9 @@ import { validateConsoleMessages } from './validateConsoleMessagesHelper';
 import { ExpectedMessage } from '../../types/expectedMessage';
 
 export const makeSpin = async (page: Page, consoleMessages: string[]) => {
+    
     await page.mouse.click(300, 300, { delay: 10 });
+    await new Promise((r) => setTimeout(r, 2000));
```
Comment (abindukalpa) [inline skyVegasTests/utilities/makeSpinHelper.ts]: Is this needed? The bug was for the search only AFAIR

Comment (cma06) [reply skyVegasTests/utilities/makeSpinHelper.ts]: Yeah,  this ticket shouldn't fix make spin function

**Review by abindukalpa — COMMENTED** [review]
Ideally the element actions should be waited on as specified by Catalin, but the timeouts are the only thing that worked for me. Tried force click and dispatch event, but that didnt give a consistent result.

**Comment by gwenso** [comment]
> We should look into these https://playwright.dev/docs/actionability rather than setting more timeouts in the code

Isn't that what is already happening with the .click() call? According to that page it should be working as we expect and waiting for everything to load but we've seen that's not the case, without manually sending the thread to sleep until everything is loaded properly

File: skyVegasTests/utilities/makeSpinHelper.ts (hunk 1 above)
Comment (Wilko2001) [reply skyVegasTests/utilities/makeSpinHelper.ts]: I noticed the issue as I was testing and thought to try and set a timeout for the spin and it worked after that. Or should this be sorted in a different ticket?

Comment (Wilko2001) [reply skyVegasTests/utilities/makeSpinHelper.ts]: I have removed this though, as we have all agreed the multiple timeouts are not the best way to sort this issue out and the ticket is just for the search bar. :)

**Comment by cma06** [comment]
That means the selectors aren't correct then or are not specific enough. We shouldn't be using awaits at all when we can target HTML elements. The only reason we are using it in the spin is because we are inside an iframe and on a canvas so we have no HTML elements to target.

**Comment by cma06** [comment]
This worked for me without awaits. But we need to wait for elements on the page to load, such as the balance amount to come through. Also I have noticed something weird in the UI when you click the search box, it overlays another search box on top of it after you click. But in anyway this shows that we don't need the await on the thread and we need to specifically wait for something on the page to happen.


import { type Page } from '@playwright/test';
import { validateConsoleMessages } from './validateConsoleMessagesHelper';

export const launchGame = async (
    page: Page,
    gameName,
    consoleMessages: string[]
) => {
    await page.waitForLoadState('documentloaded')
    
    await page.getByText('Search for games...').click();

    await page.getByPlaceholder('Search for games...').fill(gameName);
   
    await page.locator('.game-tile-from-search-component').first().click();
    await validateConsoleMessages(
        'gameLoadComplete',
        consoleMessages,
        [4_000, 6_000, 8_000, 15_000]
    );
};

**Comment by cma06** [comment]
const searchForGamesText = 'Search for games...'

    // wait for wallet amount to appear, which should ensure the page has loaded correctly
    await expect(page.locator('table.ssc-wldw tbody')).toContainText('£');

     // wait for wallet amount to appear, which should ensure the page has loaded correctly
    await page.getByText(searchForGamesText).click();

    // type the game we are searching for in the text box
    await page.getByPlaceholder(searchForGamesText).fill(gameName);

    // check that after typing in the search box, the results have appeared
    await expect(page.locator('.search-box .game-tile-from-search-component .tile-container').first()).toBeAttached()

    // click the first game result in the list
    await page.locator('.search-box .game-tile-from-search-component .tile-container').first().click();

    await validateConsoleMessages(
        'gameLoadComplete',
        consoleMessages,
    );


This should do the trick. I've ran it multiple times with no issues. Can you also double check it works with no user intervention to click the search box as well on your end ? Thanks

**Review by gwenso — APPROVED** [review]

**Pull request merged**